contributors/design-proposals/alternate-api-representations.md

```diff
@@ -0,0 +1,405 @@
+* Return a simple description of an object that can be displayed in a wide range of clients
+  (for `kubectl describe`)
+* Return the object with fields set by the server cleared (as `kubectl export`)
+
+The server should allow a common mechanism for a client to request a resource be returned
+in one of a number of possible forms.
+
+Also, the server today contains a number of objects which are common across multiple groups,
+but which clients must be able to deal with in a generic fashion. These objects - Status,
+ListMeta, ObjectMeta, List, ListOptions, ExportOptions, and Scale - are embedded into each
+group version but are actually part of a a shared API group. It must be possible for a naive
+client to translate the Scale response returned by two different API group versions.
+
+
+## Motivation
+
+Currently it is difficult for a naive client (dealing only with the list of resources
+presented by API discovery) to properly handle new and extended API groups, especially
+as versions of those groups begin to evolve. It must be possible for a naive client to
+perform a set of common operations across a wide range of groups and versions and leverage
+a predictable schema.
+
+We also foresee increasing difficulty in building clients that must deal with extensions -
+there are at least 6 known web-ui or CLI implementations that need to display some
+information about third party resources or additional API groups registered with a server
+without requiring each of them to change.  Providing a server side implementation will
+allow clients to retrieve meaningful information for the `get` and `describe` style
+operations even for new API groups.
+
+
+## Implementation
+
+The HTTP spec and the common REST paradigm provide mechanisms for clients to [negotiate
+alternative representations of objects (RFC2616 14.1)](http://www.w3.org/Protocols/rfc2616/rfc2616.txt)
+and for the server to correctly indicate a requested mechanism was chosen via the `Accept`
+and `Content-Type` headers. This is a standard request response protocol intended to allow
+clients to request the server choose a representation to return to the client based on the
+server's capabilities. In RESTful terminology, a representation is simply a known schema that
+the client is capable of handling - common schemas are HTML, JSON, XML, or protobuf, with the
+possibility of the client and server further refining the requested output via either query
+parameters or media type parameters.
+
+In order to ensure that generic clients can properly deal with many different group versions,
+we introduce the `api.k8s.io` group with version `v1` that grandfathers all existing resources
+currently described as "unversioned".  A generic client may request that responses be applied
+in this version. The contents of a particular API group version would continue to be bound into
+other group versions (`status.v1.api.k8s.io` would be bound as `Status` into all existing
```

Comment:
Why `status.v1.api.k8s.io` and not `status.api.k8s.io/v1`? Seems odd to have the version number being part of the group name, or am I missing something.